app/snapshot.go

```diff
@@ -199,13 +193,29 @@ func lsSnapshot(c *cli.Context) error {
 
 		chain, err := getChain(r.Address)
 		if err != nil {
-			return err
+			return nil, err
 		}
 
 		snapshots = util.Filter(snapshots, func(i string) bool {
 			return util.Contains(chain, i)
 		})
 	}
+	return snapshots, nil
+}
+
+func lsSnapshot(c *cli.Context) error {
```

Comment:
in `lsSnapshot`, we need to even check whether replication_factor is same as the number of replicas